src/tests/Common/CoreCLRTestLibrary/CoreClrConfigurationDetection.cs

```diff
@@ -0,0 +1,57 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+//
+
+using System;
+using System.Globalization;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Security;
+using System.Text;
+
+namespace TestLibrary;
+
+public static class CoreClrConfigurationDetection
```

Comment:
I'd really like that all these duplication is at some point somehow shared with arcade code, it would be really painful to expand this support and have it to do it twice and maybe forget or have these files go out of sync.

Reply:
For this file in particular, we can refactor this API surface out of the implementation in dotnet/arcade and use it from the Arcade package directly since this file is referenced by the tests, not the generator. For the enums, there's really not any code sharing we can do at the assembly level, but we could share at the source level if we so desired.

Reply:
I think that'd work. Can we do that? Maybe the source sharing we can do it later once this is baked in, but it would be nice to avoid duplicating this file's specific logic.

Reply:
Yes, once I finish the last part of the work I'm trying to get done here, I'll do that

Reply:
Thanks!